android/app/src/main/kotlin/app/revanced/manager/flutter/MainActivity.kt

```diff
@@ -93,29 +88,34 @@ class MainActivity : FlutterActivity() {
                 }
 
                 "getPatches" -> {
-                    val patchBundleFilePath = call.argument<String>("patchBundleFilePath")
-                    val cacheDirPath = call.argument<String>("cacheDirPath")
-
-                    if (patchBundleFilePath != null) {
-                        val patches = PatchBundleLoader.Dex(
-                            File(patchBundleFilePath),
-                            optimizedDexDirectory = File(cacheDirPath)
-                        ).map { patch ->
-                            val map = HashMap<String, Any>()
-                            map["\"name\""] = "\"${patch.patchName.replace("\"","\\\"")}\""
-                            map["\"description\""] = "\"${patch.description?.replace("\"","\\\"")}\""
-                            map["\"excluded\""] = !patch.include
-                            map["\"dependencies\""] = patch.dependencies?.map { "\"${it.java.patchName}\"" } ?: emptyList<Any>()
-                            map["\"compatiblePackages\""] = patch.compatiblePackages?.map {
-                                val map2 = HashMap<String, Any>()
-                                map2["\"name\""] = "\"${it.name}\""
-                                map2["\"versions\""] = it.versions.map { version -> "\"${version}\"" }
-                                map2
+                    val patchBundleFilePath = call.argument<String>("patchBundleFilePath")!!
+                    val cacheDirPath = call.argument<String>("cacheDirPath")!!
+
+                    val patches = PatchBundleLoader.Dex(
+                        File(patchBundleFilePath),
+                        optimizedDexDirectory = File(cacheDirPath)
+                    ).map { patch ->
+                        buildMap<String, Any> {
+                            this["\"name\""] =
+                                "\"${patch.name!!.replace("\"", "\\\"")}\""
+
+                            this["\"description\""] =
+                                "\"${patch.description?.replace("\"", "\\\"")}\""
+
+                            this["\"excluded\""] =
+                                !patch.use
```

Comment:
```suggestion
                            this["\"excluded\""] =
                                !patch.use
                            this["\"dependencies\""] =
                                patch.dependencies?.map { "\"${it.java.name}\"" }
                                    ?: emptyList<Any>()
```

Need `dependencies` as well

Reply:
Why?

Reply:
It's in use

Reply:
https://github.com/ReVanced/revanced-manager/blob/58385501881b2d606e4cad925654211c5305d56e/lib/services/patcher_api.dart#L162

here

Reply:
What use does it have?

Reply:
However the needsSettingsPatch function may be unused/unnecessary now

Reply:
https://github.com/ReVanced/revanced-manager/blob/58385501881b2d606e4cad925654211c5305d56e/lib/services/patcher_api.dart#L184

To check for dependencies and include them while patching

Reply:
I am not sure what you mean by "ability to patch from storage"

Reply:
I meant from installed, sorry

Reply:
Patching from installed will work fine if no resource patching is necessary, this is the only way to tell if resource patching is necessary

Reply:
It will only work fine, if you mount. And if you mount, resource patching will work fine anyways. Not sure why the dependencies are necessary. Also how do you check on flutter side if it's even a resource patch when you only get strings?